libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -560,6 +570,24 @@ bool AP_Arming::compass_checks(bool report)
             check_failed(ARMING_CHECK_COMPASS, report, "Compasses inconsistent");
             return false;
         }
+
+        // if ahrs is using compass and we have location, check mag field versus expected earth magnetic model
+        Location ahrs_loc;
```

Comment:
adding:
  const auto &ahrs = AP::ahrs()
would be nice

Reply:
OK, added, "AP_AHRS &ahrs = AP::ahrs()".

By the way, I could not make it const because DCM's use_compass() method actually has a side effect that may cause it to change to using GPS.  That check should really be part of DCM's update I suspect but I won't try and address that.